SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -329,51 +333,57 @@ void InterestFactory::AddOwnerInterestOnServer(Interest& OutInterest, const AAct
 	}
 }
 
-void InterestFactory::AddAlwaysRelevantAndInterestedQuery(Interest& OutInterest, const AActor* InActor, const FClassInfo& InInfo,
-														  const QueryConstraint& LevelConstraint) const
+void InterestFactory::AddClientAlwaysRelevantQuery(Interest& OutInterest, const AActor* InActor, const FClassInfo& InInfo,
+												   const QueryConstraint& LevelConstraint) const
 {
 	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
 
-	QueryConstraint AlwaysInterestedConstraint = CreateAlwaysInterestedConstraint(InActor, InInfo);
 	QueryConstraint AlwaysRelevantConstraint = CreateClientAlwaysRelevantConstraint();
 
-	QueryConstraint SystemDefinedConstraints;
-
-	if (AlwaysInterestedConstraint.IsValid())
-	{
-		SystemDefinedConstraints.OrConstraint.Add(AlwaysInterestedConstraint);
-	}
-
-	if (AlwaysRelevantConstraint.IsValid())
-	{
-		SystemDefinedConstraints.OrConstraint.Add(AlwaysRelevantConstraint);
-	}
-
 	// Add the level constraint here as all client queries need to make sure they don't check out anything outside their loaded levels.
-	QueryConstraint SystemAndLevelConstraint;
-	SystemAndLevelConstraint.AndConstraint.Add(SystemDefinedConstraints);
-	SystemAndLevelConstraint.AndConstraint.Add(LevelConstraint);
+	QueryConstraint AlwaysRelevantAndLevelConstraint;
+	AlwaysRelevantAndLevelConstraint.AndConstraint.Add(AlwaysRelevantConstraint);
+	AlwaysRelevantAndLevelConstraint.AndConstraint.Add(LevelConstraint);
 
 	Query ClientSystemQuery;
-	ClientSystemQuery.Constraint = SystemAndLevelConstraint;
+	ClientSystemQuery.Constraint = AlwaysRelevantAndLevelConstraint;
 	ClientSystemQuery.ResultComponentIds = ClientNonAuthInterestResultType.ComponentIds;
 	ClientSystemQuery.ResultComponentSetIds = ClientNonAuthInterestResultType.ComponentSetsIds;
 
 	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID, ClientSystemQuery);
+}
+
+void InterestFactory::AddAlwaysInterestedInterest(Interest& OutInterest, const AActor* InActor, const FClassInfo& InInfo) const
+{
+	QueryConstraint AlwaysInterestedConstraint = CreateAlwaysInterestedConstraint(InActor, InInfo);
 
-	// Add always interested constraint to the server as well to make sure the server sees the same as the client.
-	// The always relevant constraint is added as part of the server worker query, so leave that out here.
-	// Servers also don't need to be level constrained.
-	if (Settings->bEnableClientQueriesOnServer)
```

Comment:
where has this gone?

Reply:
This function only adds AlwaysRelevant interest now, and the server already has that, so no need to do anything special anymore.

Reply:
ah, I see, and we have changed the semantics of the AlwaysInterested to also work on servers no matter what